tests/quickcheck.rs

```diff
@@ -0,0 +1,326 @@
+extern crate num_bigint;
+extern crate num_integer;
+extern crate num_traits;
+
+#[cfg(test)]
+#[cfg(feature = "quickcheck")]
+#[macro_use]
+pub extern crate quickcheck;
+
+#[cfg(test)]
+#[cfg(feature = "quickcheck")]
+mod quickchecks {
+    use num_bigint::{BigInt, BigUint};
+    use num_traits::{Num, One, Pow, Zero};
+    use quickcheck::TestResult;
+
+    quickcheck! {
+        fn quickcheck_unsigned_eq_reflexive(a:BigUint) -> bool {
+            a == a
```

Comment:
I would be **shocked** if we ever failed a test like this, but as long as it runs quickly, I guess that's OK. Ditto for a lot of these tests that are testing broad mathematical properties.

Reply:
Yeah, I shoved the equality tests in last minute with thinking somewhere along the lines "I rely on equality a lot. I should probably test it a little first." As for the broad mathematical properties, they seem a bit silly, I agree. They really aim to catch bugs that appear for large values, but are absent for smaller ones, otherwise I could just test every operation against primitives and catch every bug. I guess if using `u128`/`i128` primitives, there is probably full branch coverage (on the one thing I'm testing) even without the broad mathematical tests, since `BigInts` are based on `Vec<u32>`. I'm going to leave them in for now, but I think new tests should mostly test against (128 bit) primitives.